octoprint_mrbeam/__init__.py

```diff
@@ -1282,16 +1274,17 @@ def custom_materials(self, data):
 
         try:
             if data.get("reset", False) == True:
-                materials(self).reset_all_custom_materials()
+                materials(self).delete_all_custom_materials()
 
-            if "delete" in data:
-                materials(self).delete_custom_material(data["delete"])
+            if data.get("delete", []):
+                for material_key in data["delete"]:
+                    materials(self).delete_custom_material(material_key)
 
-            if "put" in data and isinstance(data["put"], dict):
-                for key, m in data["put"].iteritems():
-                    materials(self).put_custom_material(key, m)
+            if isinstance(data.get("put"), dict):
```

Comment:
This was it has a fall back that is equivalent to (if "put" in data)
```suggestion
            if isinstance(data.get("put", None), dict):
```

Reply:
Actually both are the same, because the default value is None: `dict.get(key, default = None)`